### PR TITLE
Add equality check and exports

### PR DESCRIPTION
**Pull request by lbittarello**
This PR:

* adds an equality check to the distribution and link classes;
* adds exports (so downstream developers don't have to import from private modules);
* simplifies the `Union[float, int]` hints ([ref](https://stackoverflow.com/a/50928627));
* updates the pre-commit hooks (and the code accordingly).
* removes the aarch linux64 conda build from CI

### CODE REVIEW

File: .github/workflows/conda-build-linux-main.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
           - linux_64_numpy1.20python3.8.____cpython
           - linux_64_numpy1.20python3.9.____cpython
           - linux_64_numpy1.23python3.11.____cpython
-          - linux_aarch64_numpy1.21python3.10.____cpython
+          # - linux_aarch64_numpy1.21python3.10.____cpython
```
Comment (jtilly) [inline .github/workflows/conda-build-linux-main.yml:19]:
```suggestion
```

File: tests/glm/test_distribution.py [hunk 2]
```diff
@@ -85,6 +81,15 @@ def test_tweedie_distribution_parsing():
         get_family("tweedie (a)")
 
 
+def test_equality():
+    assert TweedieDistribution(1) == TweedieDistribution(1)
+    assert TweedieDistribution(1) == PoissonDistribution()
```
Comment (jtilly) [inline tests/glm/test_distribution.py:86]:
```suggestion
    assert TweedieDistribution(1) == PoissonDistribution()
    assert TweedieDistribution(2) == GammaDistribution()
```

**Review by jtilly — APPROVED** [review]
Wonderful, thanks!

Changelog entry, pretty please.

File: .flake8 [hunk 3]
```diff
@@ -1,17 +1,24 @@
 # Taken directly from https://github.com/ambv/black/blob/master/.flake8
 [flake8]
 ignore =
-    C408, # ignore temporarily https://github.com/adamchainz/flake8-comprehensions/pull/282
+    B905,
+    # ignore temporarily https://github.com/adamchainz/flake8-comprehensions/pull/282
```
Comment (MarcAntoineSchmidtQC) [inline .flake8:5]: This is not temporary anymore.
```suggestion
    # allow calling dict() instead of the literal {} for readability
```

**Review by MarcAntoineSchmidtQC — APPROVED** [review]
Great PR Luca! See my minor suggestions. Thanks!

**Pull request merged**